### PR TITLE
Airport Challenge

### PR DESCRIPTION
**Pull request by festinalent3**

### CODE REVIEW

File: spec/plane_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  #I want to prevent takeoff when weather is stormy
+
+  describe '#take_off' do
+    context 'plane can take off from airport' do
+      it 'it takes off' do
+        expect(subject.take_off).to eq true
+      end
+      it 'does not take off in stormy weather' do
+        allow(airport).to receive(:weather).and_return("stormy")
+        expect { subject.take_off }.to raise_exception "Can't take off in stormy weather"
```
Comment (festinalent3) [inline spec/plane_spec.rb]: That's just silly! ;) I prefer keeping my error message clear and specific

File: lib/plane.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @airport = airport
+  end
+
+  def land(airport)
+    @airport = airport
+    @landed = true
+  end
+
+  def take_off
+    raise "Can't take off in stormy weather" if airport.weather == "stormy"
```
Comment (festinalent3) [inline lib/plane.rb]: done!

File: lib/communicator.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    fail "Can't land in stormy weather" if airport.weather == "stormy"
+    fail "Can't land in full airport" if airport.full?
+  end
+
+  def check_can_take_off(plane, airport)
+    fail "Can't take off when flying" if plane.flying?
+    fail "Can't take off from another airport" if plane.current_airport != airport
+    fail "Can't take off in stormy weather" if airport.weather == "stormy"
+  end
+
```
Comment (danldb) [inline lib/communicator.rb]: This module seems unnecessary - remember to use fewest possible components.

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def dock(plane)
+    check_can_dock(plane)
+    update_weather
+    planes << plane
+  end
+
+  def undock(plane)
+    check_can_undock(plane)
```
Comment (danldb) [inline lib/airport.rb]: tell don't ask - here, you are asking.

File: lib/plane.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+require_relative 'communicator'
+
+class Plane
+  include Communicator
+
+  def initialize(airport = Airport.new)
+    self.landed_at(airport)
+    airport.dock(self)
+  end
```
Comment (danldb) [inline lib/plane.rb]: Plane should not be responsible for building airports.

File: lib/plane.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  def landed_at(airport)
+    @airport = airport
+    @flying = false
+    true
+  end
+
+  def is_flying
+    @airport = 'no airport'
+    @flying = true
```
Comment (danldb) [inline lib/plane.rb]: no need for `@` when using attr_accessor

**Comment by danldb** [comment]
A little over complicated. Too many components.
 Use DI instead of mixing in the weather - it reduces the coupling

File: lib/plane.rb (hunk 6 above)
Comment (festinalent3) [inline lib/plane.rb]: Yes, thanks, it was me being lazy when refactoring *bad emma*

File: lib/plane.rb (hunk 5 above)
Comment (festinalent3) [inline lib/plane.rb]: Completely agree! I wrote about it as a weird thing in the README file..

File: lib/airport.rb (hunk 4 above)
Comment (festinalent3) [inline lib/airport.rb]: Are you referring to for example method check_can_dock? Not completely sure what you mean, would really, really appreciate just a short explanation if you have the time! Thank you :)

File: lib/communicator.rb (hunk 3 above)
Comment (festinalent3) [inline lib/communicator.rb]: True, my bad, I sort of finished the challenge in its simple form at first but decided I wanted to play around with modules and that led to the decision of creating this monster for absolutely no reason at all! :D

**Pull request closed without merging**